### PR TITLE
Migrate to Hilt

### PR DESCRIPTION
**Pull request by lauriskr**
Improvements from usage feedback.

Most notably, migration to Hilt

### CODE REVIEW

**Comment by harri35** [comment]
Ran the template locally, looks good to me.

**Review by harri35 — APPROVED** [review]
Lab JIRA: P42-229
PR: https://github.com/LabMobi/project-creator-android/pull/7
It looks fine to me. It can be merged when comments are addressed.

Awesome work!

File: project/app-infrastructure/build.gradle [hunk 1]
```diff
@@ -48,6 +49,14 @@ android {
     }
 }
 
+hilt {
+    enableAggregatingTask = true
```
[the remark this answers is not in the document]
Comment (lauriskr) [reply project/app-infrastructure/build.gradle:55]: Added

File: project/app/src/main/java/mobi/lab/sample/common/ViewModelExtensions.kt [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+import androidx.lifecycle.ViewModel
+import dagger.hilt.android.lifecycle.withCreationCallback
+import kotlin.reflect.KClass
+
+/**
+ * These extensions functions help with creating ViewModels with Assisted Injection factories.
+ * There is a new CreationExtras feature that allows to pass in a callback when the ViewModel needs to be created.
+ * HiltViewModels use this feature to provide a hook to invoke the custom AssistedInjection factory interface.
+ *
+ * These extension functions are just convenience functions around the extrasProducer field and make the API less verbove.
```
[the remark this answers is not in the document]
Comment (lauriskr) [reply project/app/src/main/java/mobi/lab/sample/common/ViewModelExtensions.kt]: Fixed

**Pull request merged**